### PR TITLE
fix olm rbac and operator-tls OwnerReferences

### PR DESCRIPTION
**Pull request by dvaldivia**
This is the PR https://github.com/minio/operator/pull/788 but rebased

### CODE REVIEW

File: pkg/controller/cluster/operator.go [hunk 1]
```diff
@@ -202,9 +202,9 @@ func (c *Controller) createOperatorSecret(ctx context.Context, operator metav1.O
...
 			OwnerReferences: []metav1.OwnerReference{
 				*metav1.NewControllerRef(operator, schema.GroupVersionKind{
-					Group:   miniov2.SchemeGroupVersion.Group,
-					Version: miniov2.SchemeGroupVersion.Version,
-					Kind:    miniov2.OperatorCRDResourceKind,
+					Group:   "apps",
+					Version: "v1",
+					Kind:    "Deployment",
 				}),
 			},
```
Comment (harshavardhana) [inline pkg/controller/cluster/operator.go:209]: is it good to hardcode these values? what are the implications?

**Review by harshavardhana — CHANGES_REQUESTED** [review]

File: pkg/controller/cluster/operator.go (hunk 1 above)
Comment (dvaldivia) [reply pkg/controller/cluster/operator.go:209]: not much really, we are passing a Deployment object to this function, but it's behind the `metav1.Object` interface

Comment (Alevsk) [reply pkg/controller/cluster/operator.go:209]: Make sense, theres no operator CRD, but how does it know we want to reference the operator deployment?

Comment (dvaldivia) [reply pkg/controller/cluster/operator.go:209]: @Alevsk the `operator` reference can only be a deployment, following the code logic

**Review by harshavardhana — APPROVED** [review]

**Review by harshavardhana — APPROVED** [review]

**Review by Alevsk — APPROVED** [review]

**Pull request merged**